pkg/identity/cache/allocator.go

```diff
@@ -286,67 +286,93 @@ func (m *CachingIdentityAllocator) WaitForInitialGlobalIdentities(ctx context.Co
 	return m.IdentityAllocator.WaitForInitialSync(ctx)
 }
 
-// AllocateIdentity allocates an identity described by the specified labels. If
-// an identity for the specified set of labels already exist, the identity is
-// re-used and reference counting is performed, otherwise a new identity is
-// allocated via the kvstore or via the local identity allocator.
-// A possible previously used numeric identity for these labels can be passed
-// in as the 'oldNID' parameter; identity.InvalidIdentity must be passed if no
-// previous numeric identity exists.
-func (m *CachingIdentityAllocator) AllocateIdentity(ctx context.Context, lbls labels.Labels, notifyOwner bool, oldNID identity.NumericIdentity) (id *identity.Identity, allocated bool, err error) {
-	isNewLocally := false
+var ErrNonLocalIdentity = fmt.Errorf("labels would result in global identity")
 
-	// Notify the owner of the newly added identities so that the
-	// cached identities can be updated ASAP, rather than just
-	// relying on the kv-store update events.
-	defer func() {
-		if err == nil {
-			if allocated || isNewLocally {
-				if id.ID.HasLocalScope() {
-					metrics.Identity.WithLabelValues(identity.NodeLocalIdentityType).Inc()
-				} else if id.ID.HasRemoteNodeScope() {
-					metrics.Identity.WithLabelValues(identity.RemoteNodeIdentityType).Inc()
-				} else if id.ID.IsReservedIdentity() {
-					metrics.Identity.WithLabelValues(identity.ReservedIdentityType).Inc()
```

Comment:
note to reviewers: This case is no longer present in the new version as `allocated` could never become `true` for reserved identities